src/deadline/cinema4d_submitter/font_installer.py

```diff
@@ -0,0 +1,286 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+"""
+Utility functions for the handling of fonts
+"""
+
+import ctypes
+import logging
+import os
+import shutil
+import sys
+import traceback
+from typing import Set, Tuple
+
+if sys.platform == "win32":
+    from ctypes import wintypes
+
+    try:
+        import winreg
+    except ImportError:
+        import _winreg as winreg  # type: ignore
+
+    user32 = ctypes.WinDLL("user32", use_last_error=True)
+    gdi32 = ctypes.WinDLL("gdi32", use_last_error=True)
+else:
+    # Stub for non-Windows platforms
+    winreg = None  # type: ignore
+    user32 = None  # type: ignore
+    gdi32 = None  # type: ignore
+    wintypes = None  # type: ignore
+
+FONTS_REG_PATH = r"Software\Microsoft\Windows NT\CurrentVersion\Fonts"
+
+HWND_BROADCAST = 0xFFFF
+SMTO_ABORTIFHUNG = 0x0002
+WM_FONTCHANGE = 0x001D
+GFRI_DESCRIPTION = 1
+GFRI_ISTRUETYPE = 3
+
+INSTALL_SCOPE_USER = "USER"
+INSTALL_SCOPE_SYSTEM = "SYSTEM"
+
+FONT_LOCATION_SYSTEM = os.path.join(os.environ.get("SystemRoot", "C:\\Windows"), "Fonts")
+FONT_LOCATION_USER = os.path.join(
+    os.environ.get("LocalAppData", ""), "Microsoft", "Windows", "Fonts"
+)
+
+# Font extensions supported in gdi32.AddFontResourceW
+# OpenType fonts without an extension can also be installed (e.g. Adobe Fonts)
+FONT_EXTENSIONS = [".otf", ".ttf", ".fon", ""]
+
+logger = logging.getLogger(__name__)
+
+
+def find_fonts(session_dir: str) -> Set[str]:
```

Comment:
Could we simplify this function to just look for font files instead of searching for special directories?